src/component/2d/zones/Zone.jsx

```diff
@@ -113,7 +111,16 @@ const Zone = ({ zoneData, onDelete }) => {
           : stylesOnHover
       }
       key={id}
-      {...highlightZone.onHover}
+      {...{
+        onMouseEnter: () => {
+          assignmentZone.onMouseEnter();
+          highlightZone.show();
+        },
+        onMouseLeave: () => {
+          assignmentZone.onMouseLeave();
+          highlightZone.hide();
+        },
+      }}
```

Comment:
Why you have to use object destructuring of events object and not to use the event direct on the element?